test_iokit.py

```diff
@@ -122,3 +159,206 @@ def test_IOObjectCopySuperclassForClass():
     class_name = "IOService"
     superclass_name = iokit.corefoundation_to_native(iokit.IOObjectCopySuperclassForClass(class_name))
     assert superclass_name == "IORegistryEntry"
+
+
+def test_IOIteratorNext():
+    err, iterator = iokit.IORegistryCreateIterator (
+            iokit.kIOMainPortDefault,
+            "IOService".encode(),
+            iokit.kNilOptions,
+            None
+        )
+    assert err == 0
+
+    obj = iokit.IOIteratorNext(iterator)
+    assert obj != 0 # 0 means the iterator handle is invalid
+
+    err = iokit.IOObjectRelease(obj)
+    assert err == 0
+
+
+def test_IOIteratorIsValid():
+    err, iterator = iokit.IORegistryCreateIterator (
+            iokit.kIOMainPortDefault,
+            "IOService".encode(),
+            iokit.kNilOptions,
+            None
+        )
+    assert err == 0
+    assert iokit.IOIteratorIsValid(iterator) == True
+
+
+def test_IORegistryEntryFromPath():
+    device = iokit.IORegistryEntryFromPath (
+        iokit.kIOMainPortDefault,
+        "IOService:/IOResources".encode()
+    )
+    assert device != iokit.NULL
+
+    err = iokit.IOObjectRelease(device)
+    assert err == 0
+
+
+def test_IORegistryEntryGetLocationInPlane():
+    device = iokit.IORegistryEntryFromPath (
+        iokit.kIOMainPortDefault,
+        "IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/PEG0@1/IOPP/PEGP@0".encode()
+    )
+    assert device != iokit.NULL
```

Comment:
```py
FAILED test_iokit.py::test_IORegistryEntryGetLocationInPlane - assert 0 != 0
```

Reply:
Do you believe it's sufficient to use the following to test this?
```py
def test_IORegistryEntryGetLocationInPlane():
    device = _get_IOResources()

    err, matched = iokit.IOServiceGetMatchingServices (
            iokit.kIOMainPortDefault,
            { "IOProviderClass": "IOPCIDevice" },
            None
        )
    assert err == 0

    iters = list(iokit.ioiterator_to_list(matched))
    interface = iokit.corefoundation_to_native (
        iokit.IORegistryEntryCreateCFProperties (
            iters[0],
            None,
            iokit.kCFAllocatorDefault,
            iokit.kNilOptions
        )
    )[1]

    err, location = iokit.IORegistryEntryGetLocationInPlane (
            iters[0],
            "IOService".encode(),
            None
        )
    assert err == 0
    assert interface.get("pcidebug") != None
    assert iokit.io_name_t_to_str(location) == interface.get("pcidebug").split(":")[-1]

    for i in iters:
        err = iokit.IOObjectRelease(i)
        assert err == 0
```

Reply:
`device` is unused	+

<img width="169" alt="image" src="https://github.com/USBToolBox/tool/assets/17605561/d59b8613-8cfe-4114-b6b9-73b42bf3fb9a">
<img width="292" alt="image" src="https://github.com/USBToolBox/tool/assets/17605561/46f0bfc8-0d58-4b87-b258-83edf3c7f4e7">

Reply:
`device` has been removed, since it's no longer necessary (or so I think). How do we ensure this function is working properly, though? Perhaps by checking if the value returned from it isn't equal to `None`?